### PR TITLE
Add mod route53_record.py

### PR DESCRIPTION
**Pull request by ptux**
##### SUMMARY

I'm going to add a new module `route53_record.py`

* **support boto3**
This is why I start coding.

* **consistent module name**
This is why I name it `route53_record`.
Keep a consistency with other existing modules(route53_zone,route53_facts,route53_health_check).

* **consistent option name**
This is why I give up compatibility with existing module `route53`. 
I use options name which AWS route53 console menu uses.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
route53_record.py

##### ANSIBLE VERSION

```paste below
2.8
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/route53_record.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+RETURN = '''
+nameservers:
+  description: nameservers associated with the zone
+  returned: when state is 'get'
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/route53_record.py]: There is no state `get`.

File: lib/ansible/modules/cloud/amazon/route53_record.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        record_set_value=dict(required=False, type='list'),
+        state=dict(choices=['present', 'absent'], required=True),
+        wait=dict(required=False, type='bool', default=False),
+        wait_timeout=dict(required=False, type='int', default=300)
+    ))
+
+    required_if = [('state', 'present', ['record_set_value']),
+                   ('state', 'absent', ['record_set_value']),
+                   ('state', 'delete', ['record_set_value']),
+                   ('state', 'create', ['record_set_value'])]
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/route53_record.py]: `delete` and `create` are not valid states

**Comment by ptux** [comment]
votes and opinions, please!

**Comment by felixfontein** [comment]
You should add some working tests; the current integration tests don't do anything with the module itself.

**Comment by ptux** [comment]
@felixfontein 
I will add integration tests and options if this could be merged.
At this point, I want some opinions about the **SUMMARY**.

**Comment by willthames** [comment]
I'd rather the existing route53 module was ported to boto3 (and renamed separately) - I don't think it should be too hard to maintain backwards compatibility, and there's functionality being lost here, as far as I can tell (e.g. aliases being in a different hosted zone)

Compatability of option names with AWS is not as a high a priority, and just adds extra verbosity. One of the (few?) benefits of Ansible not maintaining consistency with AWS naming is that we can use better names!

I think this is valuable useful work that you've done, and I hope it is a good starting point for porting the existing route53 module.

**Comment by ptux** [comment]
Agree with backwards compatibility for the option name.
I could give up the new option name.

But I won't port the existing route53 module to boto3.
There is no integration test code.
I don't know how to ensure backwards compatibility.
In order to port route53 to boto3 maybe need to erase almost all the code except docstrings.

I think someone else will do.

**Comment by willthames** [comment]
Given that integration tests are required anyway for a change of this scale, it's not too hard to write a reasonable test suite that the existing module passes and then run the same test suite against the updated module. 

The test suite that would be needed doesn't need to provide 100% coverage, just enough to be reasonably confident.

**Comment by ptux** [comment]
OK, maybe you are right.
I was going to refactor route53, route53_facts, route53_health_check.
However, I couldn't refactor in your way.
It seems that some others have refactored the other modules of aws in your way.
So, I'll change back to use AWS CLI until someone refactors them.

**Pull request closed without merging**